src/MongoMemoryServer.ts

```diff
@@ -39,7 +39,8 @@ export interface MongoInstanceDataT {
   replSet?: string;
 }
 
-async function generateConnectionString(port: number, dbName: string): Promise<string> {
+// TODO: do we need to keep this function async ?
+const generateConnectionString = async (port: number, dbName: string): Promise<string> => {
```

Comment:
@nodkz Could we remove the async part to this function ?

Reply:
Nope. Connection string can be obtained only if the server is launched and ready.

Reply:
@nodkz Well this function is only called if we have a **port** and a **dbName** coming from the launched server, so I don't really understand how the async part here could assure us the server is launched and ready since it only returns a string without any check. 
I would understand that before calling this function we should assure that the server is launched and ready, what we did here for example with the `await this.getInstanceData` before calling `generateConnectionString` : 
```javascript
const { uri, port }: MongoInstanceDataT = await this.getInstanceData();

    // IF true OR string
    if (otherDbName) {
      if (typeof otherDbName === 'string') {
        // generate uri with provided DB name on existed DB instance
        return generateConnectionString(port, otherDbName);
      }
      // generate new random db name
      return generateConnectionString(port, generateDbName());
    }
```

Reply:
See deeper
```js
async getInstanceData(): Promise<MongoInstanceDataT> {
    if (!this.runningInstance) {
      await this.start();
    }
    if (this.runningInstance) {
      return this.runningInstance;
    }
    throw new Error(
      'Database instance is not running. You should start database by calling start() method. BTW it should start automatically if opts.autoStart!=false. Also you may provide opts.debug=true for more info.'
    );
  }
```

When instance not started, this method call `start` and return proper connection string.
Proper connection string cannot be obtained in sync, cause may be busy port and `start()` will try to use another port.

Reply:
@nodkz Ok I didn't saw that :) I'll remove the TODO.